miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/userinfo/UserInfoHandler.kt

```diff
@@ -0,0 +1,104 @@
+package com.rakuten.tech.mobile.miniapp.js.userinfo
+
+import com.rakuten.tech.mobile.miniapp.MiniAppSdkException
+import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridgeListener
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionCache
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionResult
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionType
+
+/**
+ * A class to provide the interfaces for getting user info e.g. user-name, profile-photo etc.
+ */
+@Suppress("TooGenericExceptionCaught", "LongMethod")
+open class UserInfoHandler {
```

Comment:
Should we prevent the Host App from directly instantiating this class? We could either make it `abstract class` or add an `internal constructor`?